### PR TITLE
Adding dockerization

### PR DESCRIPTION
**Pull request by ashok-an**
## Description

Adding Dockerfile support to help build a container image. Updated documentation and did some minor update to the `curl` command

## Type of Change

- [ ] Bug Fix
- [*] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [*] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [*] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,7 +35,8 @@ You can do all of that in the command line using [curl](https://curl.se/) like t
 ```shell
 mkdir my-test-telemetry
 cd my-test-telemetry
-curl -O -L https://github.com/cisco-open/test-telemetry-generator/releases/latest/download/test-telemetry-generator-otel-proto-0.18.0-fatjar.jar
+version=v23.7.0
+curl -O -L https://github.com/cisco-open/test-telemetry-generator/releases/download/${version}/test-telemetry-generator-otel-proto-${version}-fatjar.jar
```
Comment (svrnm) [inline README.md:39]: good catch, but maybe this should be in a separate PR.

File: README.md [hunk 2]
```diff
@@ -82,6 +83,16 @@ You now have a working setup from which you can build your own telemetry test en
 
 ***
 
+## Docker build
+```text
+# Download the files
+# Update the .yaml manifests
+# build
+jarFile=$(ls -1 *.jar)
+imageName="otel-test-image"
```
Comment (svrnm) [inline README.md:92]: Wouldn't it be easier to include the `curl` of the fat jar in the docker build? this way it is self contained. As a next step we could think about building it and publishing it via ghcr.io maing it even easier for people to consume it

File: README.md (hunk 1 above)
Comment (ashok-an) [reply README.md:39]: may be do you want me to remove it?

File: README.md (hunk 2 above)
Comment (ashok-an) [reply README.md:92]: Wanted to keep it simple; then we would also need to think of proxies, Docker-in-Docker situation to be complete right?

File: README.md (hunk 1 above)
Comment (svrnm) [reply README.md:39]: yes, remove it (and raise a new PR with only that change)

**Comment by ashok-an** [comment]
Pushed a new version with updated README, the volume mount option is not working for some reason (could be my JRE container doesn't have proper shell)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ENV PATH="$PATH:$JAVA_HOME/bin"
+
+WORKDIR /home
+
+ARG jarFile
+ENV JARFILE=${jarFile}
+
+COPY $jarFile /home
+COPY *.yaml /home
+ENTRYPOINT java -jar ${JARFILE} -r resource-definition.yaml -s trace-definition.yaml -t cli-target-rest.yaml
```
Comment (preet-dev) [inline Dockerfile:32]: Any dir paths need to be stripped off of the JARFILE env var.

**Review by preet-dev — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (ashok-an) [reply Dockerfile:32]: It wont be right? due to COPY command? if it is wrong COPY will fail?
Any specific use-cases?

Comment (preet-dev) [reply Dockerfile:32]:
```
test-telemetry-generator % gw clean build jar fatJar -x test

test-telemetry-generator % docker build . -t test_ttg --build-arg jarFile=build/libs/test-telemetry-generator-all.jar 

test-telemetry-generator % docker run test_ttg
Error: Unable to access jarfile build/libs/test-telemetry-generator-all.jar
```

**Comment by preet-dev** [comment]
@ashok-an @svrnm I've opened another PR on top of these changes. The build process is fairly straightforward and we don't need a jarFile param. Also added CI to make sure future commits don't break docker build.

**Comment by preet-dev** [comment]
@ashok-an @svrnm I've opened another PR on top of these changes. The build process is fairly straightforward and we don't need a jarFile param. Also added CI to make sure future commits don't break docker build.
